### PR TITLE
add equal method

### PR DESCRIPTION
**Pull request by nicolasparada**
Closes https://github.com/calyptia/go-fluentbit-config/issues/52

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -4,6 +4,9 @@ import (
 	"fmt"
 	"strings"
 
+	"golang.org/x/exp/maps"
```
Comment (pwhelan) [inline config.go:7]: I'm a bit concerned about using `exp` packages. I was unable to gloss anything about their status. The functionality they provide can probably just be re-implemented quickly if needed. Add some tests that actually use the functionality explicitly, ie:

- Equality where the map is defined in a different order
- Slices in different order

Just to make sure we catch a few of the corner cases they are meant to handle and I think that should be enough to cover the risk there.

**Review by pwhelan — CHANGES_REQUESTED** [review]

File: config.go (hunk 1 above)
Comment (nicolasparada) [reply config.go:7]: I added tests for the slices.
The map... well the thing is that config should not have a map with more than one key. This map is single key.

Array -> single key map -> properties.

Otherwise you cannot define the same plugin (input, parser, filter or output) multiple times.

Comment (pwhelan) [reply config.go:7]: I was trying to think of a case where we need to compare maps directly but none occur to me for now.

**Review by pwhelan — APPROVED** [review]
